File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        ans = []

        map = {}

        for i in nums:
            if i in map:
                map[i] = map[i] + 1
            else:
                map[i] = 1


        for key in map:
            if map[key] == 2:
                ans.append(key)


        return ans                   

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        ArrayList<Integer> ansarr = new ArrayList<>();

        HashMap<Integer,Integer> mpp = new HashMap<>();

        for(int i =0;i<nums.length;i++)
        {
            if(mpp.containsKey(nums[i]))
            {
                int temp = mpp.get(nums[i]);
                mpp.put(nums[i], temp+1);
            }else{
                mpp.put(nums[i], 1);
            }
        }

        for(Integer i : mpp.keySet())
        {
            if(mpp.get(i) == 2)
            {
                ansarr.add(i);
            }
        }

        return ansarr; 
    }
}